#! /bin/sh /usr/share/dpatch/dpatch-run
## 50fix_next_outside_loop_with_return.dpatch by Niko Tyni <ntyni@iki.fi>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Exit subroutine with 'return' instead of 'next'.
## DP: David Sobon <dsobon@gmail.com>, #437474

@DPATCH@
diff -urNad libnet-irc-perl~/DCC.pm libnet-irc-perl/DCC.pm
--- libnet-irc-perl~/DCC.pm	2007-09-16 20:48:27.000000000 +0300
+++ libnet-irc-perl/DCC.pm	2007-09-16 20:51:13.000000000 +0300
@@ -253,7 +253,7 @@
 
     my $line = $self->_getline($_[0], 'BLOCKS');
 
-    next unless defined $line;
+    return unless defined $line;
     unless(print {$self->{_fh}} $line) {
 	carp ("Error writing to " . $self->{_filename} . ": $!");
 	close $self->{_fh};
